task/git-cli/0.3/README.md

```diff
@@ -0,0 +1,154 @@
+# Git Task
+
+This `Task` is Git task to work with repositories used by other tasks
+in your Pipeline.
+
+## `git-cli`
+
+This [task](../0.3/git-cli.yaml) can be used to perform `git operations`.
+All git commands can be found [here](https://git-scm.com/docs).
+
+Command that needs to be run can be passed as a script to the task.
+
+### Workspaces
+
+* **source**: A workspace that contains the fetched git repository.
+* **input**: A workspace that contains file that needs to be added to git.
+* **ssh-directory**: An optional workspace to provide SSH credentials. At
+  minimum this should include a private key but can also include other common
+  files from `.ssh` including `config` and `known_hosts`. It is **strongly**
+  recommended that this workspace be bound to a Kubernetes `Secret`.
+  For details on the correct format of the files in this Workspace
+  see [Using SSH credentials](#using-ssh-credentials) below.
+
+* **basic-auth**: An optional workspace containing `.gitconfig` and
+  `.git-credentials` files. This allows username/password/access token to be
+  provided for basic auth.
+
+  It is **strongly** recommended that this workspace be bound to a Kubernetes
+  `Secret`. For details on the correct format of the files in this Workspace
+  see [Using basic-auth Credentials](#using-basic-auth-credentials) below.
+
+### Parameters
+
+* **BASE_IMAGE**: The base image for the task.
+(_default_: `alpine/git:latest`)
+* **GIT_USER_NAME**: Git user name for performing git operation.
+* **GIT_USER_EMAIL**:  Git user email for performing git operation.
+* **GIT_SCRIPT**: The git script to run. (_required_)
+* **VERBOSE**: Log the commands that are executed during `git-cli`'s operation. (_default_: true)
+* **USER_HOME**: The user's home directory. Set this explicitly if you are running the image as a non-root user. (_default_: "/root")
+
+### Results
+
+* **commit**: The precise commit SHA after git operation is performed.
+
+### Platforms
+
+The Task can be run on `linux/amd64`, `linux/s390x` and `linux/ppc64le` platforms.
+
+### Usage
+
+This task needs authentication to git in order to push after the git operation.
+
+After creating the task, you should now be able to execute `git` commands by
+specifying the command you would like to run as the `GIT_SCRIPT` param.
+
+`Example`:
+
+```yaml
+  params:
+    - name: GIT_SCRIPT
+      value: |
+        git init
+        git remote add origin https://github.com/kelseyhightower/nocode
+        git pull origin master
+```
+
+## Using SSH credentials
+
+This Task supports fetching private repositories using SSH credentials.
+
+1. Bind an `ssh-directory` workspace to this Task.
+The workspace should contain private keys (e.g. `id_rsa`), `config`
+and `known_hosts` files - anything you need to interact with your git remote
```

Comment:
Can you please add a guide how to obtain `id_rsa`, `config` and `known_hosts`? (I know that it was not there earlier but if possible can you add? :sweat_smile: )

Reply:
Hm, isn't a primer on SSH a bit out of scope for this task? Or do you mean just a link to an article like https://www.digitalocean.com/community/tutorials/ssh-essentials-working-with-ssh-servers-clients-and-keys?

Reply:
yeah link to the article sounds good :+1: